packager/VisibilityChecker.cc

```diff
@@ -474,13 +510,26 @@ class VisibilityCheckerPass final {
                                            "# packaged: false");
                         }
                     }
-                } else if (importedAsTest) {
+                } else if (testImportInProd) {
                     ENFORCE(!isTestImport);
                     if (auto e = ctx.beginError(lit.loc(), core::errors::Packager::UsedTestOnlyName)) {
                         e.setHeader("Used `{}` constant `{}` in non-test file", "test_import", litSymbol.show(ctx));
                         e.addErrorLine(pkg.declLoc(), "Defined here");
                         auto &pkg = ctx.state.packageDB().getPackageInfo(otherPackage);
-                        if (auto exp = this->package.addImport(ctx, pkg, false)) {
+                        if (auto exp = this->package.addImport(ctx, pkg, core::packages::ImportType::Normal)) {
+                            e.addAutocorrect(std::move(exp.value()));
+                            if (!db.errorHint().empty()) {
+                                e.addErrorNote("{}", db.errorHint());
+                            }
+                        }
+                    }
+                } else if (testUnitImportInHelper) {
+                    if (auto e = ctx.beginError(lit.loc(), core::errors::Packager::UsedTestOnlyName)) {
+                        e.setHeader("The `{}` constant `{}` can only be used in `{}` files", "test_import",
```

Comment:
The error message should probably specify that this is a `test_import only:` somehow, so that it's obvious how it's different from the case in `testImportInProd`. We should also make the phrasing consistent across the 2 messages if we can.